File: Python/Q4_Heranca.py
class Animal:
    def __init__(self,especie,nome):
        self.especie = especie
        self.nome = nome

    def emitir_som(self):
        pass

class Cachorro(Animal):
    def emitir_som(self):
        return "Au Au"

class Gato(Animal):
    def emitir_som(self):
        return "Miau"

bicho = Cachorro(Cachorro, "Bicho")
bicho2 = Gato(Gato, "Bichano")
print(bicho2.emitir_som())
print(bicho.emitir_som())

File: go/Q4_Heranca/Q4_Heranca.go
package main

import (
	"fmt"
)

type Animal struct {
	nome    string
	especie string
}



func (a Animal) Som() string {
	return ""
}

type Cachorro struct {
	Animal 
}

func (c Cachorro) Som() string {
	return "Au Au"
}

type Gato struct {
	Animal
}

func (g Gato) Som() string {
	return "Miau"
}


func main() {
	cachorro := Cachorro{Animal: Animal{nome:"Bicho",especie: "Cachorro"},}

	gato := Gato{Animal: Animal{nome:"Bichano",especie: "Gato"},}

	fmt.Println(cachorro.Som())
	fmt.Println(gato.Som())
}
